aptos-move/framework/supra-framework/sources/automation_registry.move

```
@@ -370,22 +454,25 @@ module supra_framework::automation_registry {
         refund_automation_task_fee(
             signer::address_of(owner),
             &automation_task_metadata,
-            automation_registry.automation_unit_price
+            &automation_registry_config.main_config
         );
     }
 
     /// Refunds the automation task fee to the user who has removed their task registration from the list.
     fun refund_automation_task_fee(
         user: address,
         automation_task_metadata: &AutomationTaskMetaData,
-        automation_unit_price: u64,
+        automation_registry_config: &AutomationRegistryConfig,
     ) acquires AutomationRegistry {
         let current_time = timestamp::now_seconds();
-        let expiry_time_duration = automation_task_metadata.expiry_time - current_time;
+        assert!(automation_task_metadata.expiry_time > current_time, ETASK_IS_ALREADY_EXPIRED);
```

I do not think we should report an error here, we simply should not do any refund.